tests/runtimes/simple-keyvalue/src/lib.rs

```diff
@@ -0,0 +1,270 @@
+//! Simple keyvalue runtime.
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use oasis_runtime_sdk::{
+    self as sdk,
+    context::{DispatchContext, TxContext},
+    core::common::{cbor, version::Version},
+    error::Error as _,
+    module::{CallableMethodInfo, Module as _},
+    types::transaction::CallResult,
+};
+
+/// The name of our module.
+const MODULE_NAME: &str = "keyvalue";
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Key {
+    #[serde(rename = "key")]
+    #[serde(with = "serde_bytes")]
+    key: Vec<u8>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct KeyValue {
+    #[serde(rename = "key")]
+    #[serde(with = "serde_bytes")]
+    key: Vec<u8>,
+
+    #[serde(rename = "value")]
+    #[serde(with = "serde_bytes")]
+    value: Vec<u8>,
+}
+
+/// Errors emitted by the keyvalue module.
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid argument")]
+    InvalidArgument,
+}
+
+impl sdk::error::Error for Error {
+    fn module(&self) -> &str {
+        MODULE_NAME
+    }
+
+    fn code(&self) -> u32 {
+        match self {
+            Error::InvalidArgument => 1,
+        }
+    }
+}
+
+impl From<Error> for sdk::error::RuntimeError {
+    fn from(err: Error) -> sdk::error::RuntimeError {
+        sdk::error::RuntimeError::new(err.module(), err.code(), &err.msg())
+    }
+}
+
+/// Events emitted by the keyvalue module (none so far).
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Event {
+    // XXX: What if we don't want to have any events?  The SDK seems to require an implementation anyways :/
```

Comment:
Make an issue or PR against the SDK to support modules without events (e.g., where the event type is `()`). Also you probably want to test events as well.

Reply:
Filed as #61.